pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/storage/HadoopPinotFS.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.storage;
+
+import com.google.common.base.Strings;
+import com.linkedin.pinot.common.segment.fetcher.HdfsSegmentFetcher;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import org.apache.commons.configuration.Configuration;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.*;
```

Comment:
It may make sense to pull the constants out of SegmentFetcher if they are more broadly applicable.